lib/searchkick/query.rb

```diff
@@ -275,35 +275,62 @@ def prepare
             }
           end
 
-          if conversions_field && options[:conversions] != false
-            # wrap payload in a bool query
-            script_score =
-              if below12?
-                {script_score: {script: "doc['count'].value"}}
-              else
-                {field_value_factor: {field: "#{conversions_field}.count"}}
-              end
+          if conversions_fields.present? && options[:conversions] != false
+            shoulds = []
+            conversions_fields.each do |conversions_field|
+              # wrap payload in a bool query
+              script_score =
+                if below12?
+                  {script_score: {script: "doc['count'].value"}}
+                else
+                  {field_value_factor: {field: "#{conversions_field}.count"}}
+                end
 
-            payload = {
-              bool: {
-                must: payload,
-                should: {
-                  nested: {
-                    path: conversions_field,
-                    score_mode: "sum",
-                    query: {
-                      function_score: {
-                        boost_mode: "replace",
-                        query: {
-                          match: {
-                            "#{conversions_field}.query" => term
+              shoulds <<  {
+                            nested: {
+                              path: conversions_field,
+                              score_mode: "sum",
+                              query: {
+                                function_score: {
+                                  boost_mode: "replace",
+                                  query: {
+                                    match: {
+                                      "#{conversions_field}.query" => term
+                                    }
+                                  }
+                                }.merge(script_score)
+                              }
+                            }
                           }
-                        }
-                      }.merge(script_score)
+
+
+              payload = {
```

Comment:
@meetrajesh what's the reason for changing the payload here vs adding the shoulds below?